syft/execution/placeholder.py

```diff
@@ -157,6 +168,36 @@ def create_from(tensor, owner, role=None, tracing=False):
         """
         return PlaceHolder(owner=owner, role=role, tracing=tracing).instantiate(tensor)
 
+    @staticmethod
+    def insert(tensor, after, owner, role=None, tracing=False):
```

Comment:
Just a side comment:
With these, are Placeholders now Tracers? ^^

Reply:
This part seems weird to me too. Starting to think `Placeholder` has too many responsibilities, and maybe we should split placeholding and tracing into separate classes.

Reply:
Hmm, why Tracers? `.insert` is similar to `.create_from`, with extra arg to create Placeholder deeper in chain.

Reply:
Because `Placeholders` weren’t designed to go deeper in the chain, they were supposed to hold the place of a tensor chain. Putting them deeper in the chain makes tracing work, but changes how the class is used in a way that makes the name inaccurate.

Reply:
(To be clear, I’m still okay with merging this and resolving the naming issue later, but I do see an issue. It might even be the one @Jasopaum is referring to.) :smile:

Reply:
I agree with what was said here 🙂